### PR TITLE
Updated Data Model for Users and Posts types

### LINKED ISSUE #21: Find and Add Anonymous Flags/Name to Post Model
Add isAnon flag and anon names for Post mode.

Similar to #13 but could be useful for anonymity within posts.

### PR DESCRIPTION
**Pull request by xavier-xia-99**
1. `src/types/post.ts`

2. `src/types/user.ts` 

Others are comments and places where display_names are changed.

This resolves #21 

=====================================================================

Sprint 2: 
Modified `openapi` schema definitions for respective post and topics

### CODE REVIEW

**Comment by xavier-xia-99** [comment]
Added bool flag and anonname for `post` data model as we are using that for the plugin. @PeterNotFound114

**Comment by xavier-xia-99** [comment]
PR Edit : Added modifications to the scheme under `openapi` including postObject, topicObject and pid which should be needed for IF-ELSE in the respective `.tpl` files and to propagate changes from `composer.js`

File: public/openapi/components/schemas/PostObject.yaml
Comment (PeterNotFound114) [inline public/openapi/components/schemas/PostObject.yaml:1]: We possibly only need isAnon for checking if the user elects to post as anonymous. The displayed name can be changed in the .tpl files

File: src/user/data.js
Comment (PeterNotFound114) [inline src/user/data.js:1]: I think we only need to display as anon while still storing the user who posted anonymously. Displayname might be beyond our scope as of now.

**Review by PeterNotFound114 — CHANGES_REQUESTED** [review]
I suggest deleting the "anonname" field before merging. We can do changes in the .tpl files in another commit.

**Comment by xavier-xia-99** [comment]
> I suggest deleting the "anonname" field before merging. We can do changes in the .tpl files in another commit.

Should we not have a field to provide names for users, considering we might want to have a pool of anonymous names? I guess we could defer that to later too!

**Review by PeterNotFound114 — APPROVED** [review]
LGTM. I shall edit the corresponding tpl files in the next commit for the anonymous posts to show.

**Pull request merged**